httpbin/cmd/cmd_test.go

```diff
@@ -500,7 +557,25 @@ func TestMainImpl(t *testing.T) {
 				"-host", "127.0.0.1", // default of 0.0.0.0 causes annoying permission popup on macOS
 			},
 			wantCode: 1,
-			wantOut:  "go-httpbin listening on http://127.0.0.1:-256\nerror: listen tcp: address -256: invalid port\n",
+			wantOut:  "level=INFO msg=\"go-httpbin listening on http://127.0.0.1:-256\"\nlevel=ERROR msg=\"error: listen tcp: address -256: invalid port\"\n",
+			wantOutFn: func(wantOut string, out string) bool {
+				wantLines := strings.Split(strings.TrimSpace(wantOut), "\n")
+
+				var patternParts []string
+				for _, line := range wantLines {
+					escapedLine := regexp.QuoteMeta(line)
+					patternParts = append(patternParts, `time=\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+[+-]\d{2}:\d{2} `+escapedLine)
+				}
+
+				pattern := `(?m)^` + strings.Join(patternParts, `\n`)
+
+				matched, err := regexp.MatchString(pattern, out)
+				if err != nil {
+					fmt.Printf("Regex error: %v\n", err)
+					return false
+				}
+				return matched
+			},
```

Comment:
Apologies for not being clear enough earlier, but I think this is more complex than is necessary.  I think each test case should specify either `wantOut` OR `wantOutFn`, but not both.  This test might be simplified down to something like:

```go
wantOutFn: func(t *testing.T, out string) bool {
    assert.Contains(t, out, `msg="error: listen tcp: address -256: invalid port"`, "error log"`)
},
```

I think that would accomplish the same thing but much more clearly.  (I'm using go-httpbin's own [internal assert package helper](https://github.com/mccutchen/go-httpbin/blob/main/internal/testing/assert/assert.go#L95-L101) here.)

--

This obviously implies a change to the `wantOutFn` signature in the testcase struct definition, and to the behavior of the actual test code calling the function, and to the similar testcase below.

I hope that makes sense overall!

Reply:
Yeah it makes sense, it was overcomplex which is not desirable, thanks a lot for showing me this way